python/GafferUI/NodeGraph.py

```diff
@@ -52,11 +52,16 @@ def __init__( self, scriptNode, **kw ) :
 			] ),
 		)
 
+		layoutNodes =  not self.__nodesHaveUIPosition( scriptNode )
```

Comment:
I presume you're having to compute this here rather than later because of the `// place it if it's not placed already.` code [here](https://github.com/ImageEngine/gaffer/blob/master/src/GafferUI/GraphGadget.cpp#L1379). I'm more than happy to lose that code if it helps with this.